File: LCA_BST.java
//O(height)
public TreeNode LCA_BST(TreeNode root, TreeNode node1, TreeNode node2){
    if(root == null)
        return null;
    if(root.value < node1.value && root.value < node2.value)
        return LCA_BST(root.right, node1, node2);
    if(root.value > node1.value && root.value > node2.value)
        return LCA_BST(root.left, node1, node2);
    return root;
}

File: tree/LCA/LCA_BST.py
# O(height)
def LCA_bst(root, node1, node2):
    if root is None:
        return root
    if root.value < node1.value and root.value < node2.value:
        return LCA_bst(root.left, node1, node2)
    if root.value > node1.value and root.value > root2.value:
        return LCA_bst(root.right, node1, node2)
    return root
    
